drivers/hmis/app/models/hmis/form/definition.rb

```diff
@@ -57,6 +59,132 @@ def self.validate_json(json)
     recur_check.call(json)
   end
 
+  def hud_assessment?
+    Types::HmisSchema::Enums::AssessmentRole.as_data_collection_stage(role) != 99
+  end
+
+  def intake?
+    Types::HmisSchema::Enums::AssessmentRole.as_data_collection_stage(role) == 1
+  end
+
+  def exit?
+    Types::HmisSchema::Enums::AssessmentRole.as_data_collection_stage(role) == 3
+  end
+
+  def assessment_date_item
+    @assessment_date_item ||= link_id_item_hash.values.find(&:assessment_date)
+  end
+
+  def find_and_validate_assessment_date(hud_values:, entry_date:, exit_date:)
```

Comment:
This function has the logic for determining and validating the assessment date.

Currently it validates that it is on or after the entry date, and on or before the exit date.